src/module/s3.js

```diff
@@ -22,21 +21,15 @@ module.exports.S3 = ({
       await sleep(backoffFunction(count));
       try {
         // eslint-disable-next-line no-await-in-loop
-        return await call(action, opts, { expectedErrorCodes });
+        return await call(action, opts, {
+          expectedErrorCodes,
+          meta: { retryCount: count }
+        });
       } catch (e) {
-        if (get(e, 'code') === 'SlowDown') {
-          lastError = e;
-          if (logger !== null) {
-            logger.warn(`Failed to submit ${action}\n${JSON.stringify({
-              errorCode: get(e, 'code'),
-              action,
-              opts,
-              retryCount: count
-            })}`);
-          }
-        } else {
+        if (get(e, 'code') !== 'SlowDown') {
```

Comment:
```suggestion
        if (
        get(e, 'code') !== 'SlowDown'
        || count === maxRetries
        ) {
```
and remove lastError entirely?